### ISSUE TITLE
Introduce range-over-func `iter.Seq` method `All()` for `set.Set`

### ISSUE THREAD

**Issue by jbduncan** [jbduncan/go-containers#6]
Blocked on the currently-experimental [range-over-func](https://github.com/golang/go/issues/61405) Go feature being merged into a stable version of Go.

Can be satisfied by completing https://github.com/jbduncan/go-containers/issues/8.

This will allow sets to be passed into `for` loops without needing `Set.ForEach`, which doesn't support early exits from the loop. Furthermore, it supports turning the set into a "pull iterator" with the proposed `iter.Pull`, which lets user control when the iterations happen.

**Comment by jbduncan** [comment]
This was done a while ago, so can be closed.
